ctapipe_io_lst/calibration.py

```diff
@@ -556,100 +556,104 @@ def interpolate_spike_A(waveform, position):
 
 
 @njit(cache=True)
-def get_spike_A_positions(current_fc, last_fc):
+def get_spike_A_positions_base(current_first_cap, previous_first_cap, shift):
     '''
-    Find spike positions for the firmware after run `LAST_RUN_WITH_OLD_FIRMWARE`
+    Find spike positions.
+
+    For the new firmware, use shift=1, for the old firmware shift=2.
 
     Parameters
     ----------
-    fc : ndarray
+    current_first_cap: ndarray
         First capacitor of the current event
-    fc_old : ndarray
+    previous_first_cap: ndarray
         First capacitor of the previous event
 
     Returns
     -------
-    positions: List[int]
+    positions: list[int]
         List of spike positions
     '''
-    LAST_IN_FIRST_HALF = N_CAPACITORS_CHANNEL // 2 - 1
+    last_in_first_half = N_CAPACITORS_CHANNEL // 2 - 1
+    last_capacitor = (previous_first_cap + N_SAMPLES - 1) % N_CAPACITORS_CHANNEL
+
+    # The correction is only needed for even last capacitor
+    # in the first half of the DRS4 ring
+    if last_capacitor % 2 != 0 or last_capacitor > last_in_first_half:
+        # bad trickery to get numba to compile an empty list with type int
+        # see https://numba.pydata.org/numba-doc/latest/user/troubleshoot.html
+        return [int(x) for x in range(0)]
+
+    # we have two cases for spikes that can occur in each of the 4 channels
+    base_positions = (
+        N_CAPACITORS_PIXEL - N_SAMPLES - previous_first_cap - shift,
+        previous_first_cap + N_SAMPLES - shift
+    )
```

Comment:
I'd suggest using `last_capacitor` defined above to make it easier to understand.
```
best_positions = (
    N_CAPACITORS_PIXEL - last_capacitor - 2 - shift,
    last_capacitor - shift
)
```
and `shift` is 0, 1 for the current and old firmware respectively.
What do you think？

Reply:
Perfect